Data/HashMap/Base.hs

```diff
@@ -354,9 +354,11 @@ instance Ord k => Ord1 (HashMap k) where
 
 -- | The order is total.
 --
--- /Note:/ Because the hash is not guaranteed to be stable across library
--- versions, OSes, or architectures, neither is an actual order of elements in
--- 'HashMap' or an result of `compare`.is stable.
+-- /Note:/ The ordering of elements in 'HashMap' depends primarily on the keys'
+-- hashes. Therefore, it inherits any of the properties of the hashes. In
+-- particular, if the hash in use is inconsistent between library versions,
+-- OSes or architectures, then the ordering will be inconsistent in the same
+-- circumstance.
```

Comment:
Proposing a language change that makes it clear in both yours and @sjakobi's suggestion: 
 
```suggestion
-- OSes or architectures, then the ordering will reflect these inconsistencies. 
```